subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningCategory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems;
+
+import org.gradle.api.Incubating;
+
+/**
+ * {@link Problem} instance builder requiring the specification of the problem type.
+ *
+ * @since 8.5
+ */
+@Incubating
+public interface ProblemBuilderDefiningCategory {
+
+    /**
+     * Declares the problem category string.
```

Comment:
❌ I the category is a type with string representation, which is an implementation detail imho.
```suggestion
     * Declares the problem category.
```